### PR TITLE
fix: allow unicode page names

### LINKED ISSUE #2888: JS execution silently fails on a statically generated site
JS execution silently fails on a statically generated site when a route path contains non-ascii characters.

Steps to reproduce:
```
vue init nuxt-community/starter-template nuxt-project
cd nuxt-project
npm i
cp pages/index.vue pages/провизорам.vue
npm run generate
npm i http-server
npx http-server dist
```
Put some console.logs into провизорам.vue and index.vue, for example
```
<template>
  <div></div>
</template>

<script>
export default {
  mounted () {
    console.log('test')
  }
}
</script>
```

Visit http://localhost:8080/провизорам and you'll see that the console.log does not get executed and that there is no error in the developer console. But if you put some code into index.vue and visit http://localhost:8080/ the code will execute.

When running `npm run dev` this problem does not manifest itself.

<div align="right"><sub><em>This question is available on <a href="https://nuxtjs.cmty.io">Nuxt.js</a> community (<a href="https://nuxtjs.cmty.io/nuxt/nuxt.js/issues/c2510">#c2510</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by aldarund**
## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Resolves: #2888 #3211

## Checklist:
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [x] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests passed.

### CODE REVIEW

File: test/fixtures/spa/nuxt.config.js [hunk 1]
```diff
@@ -10,6 +10,9 @@ export default {
     }
   },
   build: {
+    optimization: {
+      minimize: false
```
Comment (manniL) [inline test/fixtures/spa/nuxt.config.js]: Is this change necessary?

File: packages/vue-app/package.json [hunk 2]
```diff
@@ -3,13 +3,13 @@
...
   "repository": "nuxt/nuxt.js",
   "license": "MIT",
-  "typings": "types/index.d.ts",
   "files": [
     "dist",
     "template",
     "types/*.d.ts"
   ],
   "main": "dist/vue-app.js",
+  "typings": "types/index.d.ts",
```
Comment (manniL) [inline packages/vue-app/package.json]: Is this change necessary?

File: test/fixtures/spa/nuxt.config.js (hunk 1 above)
Comment (aldarund) [reply test/fixtures/spa/nuxt.config.js]: oh, no :) debug leftovers

File: packages/vue-app/package.json (hunk 2 above)
Comment (aldarund) [reply packages/vue-app/package.json]: @manniL not sure about this. Its done automatically when i do just do build, so i guess yes, unless its done inverse for someone other :)

Comment (pi0) [reply packages/vue-app/package.json]: My bad when merging original typings PR. Our home-made pkg script sorts fields inside `package.json` so I confirm it is valid.

**Review by TheAlexLichter — APPROVED** [review]

**Review by pi0 — APPROVED** [review]
